R/conclass.R

```diff
@@ -408,42 +437,55 @@ Conos <- R6::R6Class("Conos", lock_objects=F,
     },
 
     #' @description  Generate an embedding of a joint graph.
-
+    #'
     #' @param method embedding method. Currently largeVis and UMAP are supported
+    #' @param name optional name of the embedding result (will default to the method name; otherwise, "embedding")
     #' @param M, gamma, alpha, sgd__batched - largeVis parameters (defaults are 1, 1, 0.01, 1e8 respectively).
     #' @param perplexity perplexity passed to largeVis (defaults to NA).
     #' @param seed random seed for the largeVis algorithm. Default: 1.
     #' @param target.dims numer of dimensions for the reduction. Default: 2. Higher dimensions can be used to generate embeddings for subsequent reductions by other methods, such as tSNE
     #' @param n.cores number of cores, overrides class field
     #' @param verbose verbose mode. Default: TRUE.
     #' @param ... additional arguments, passed to UMAP embedding (run ?conos:::embedGraphUmap for more info)
-    embedGraph=function(method='largeVis', M=1, gamma=1, alpha=0.1, perplexity=NA, sgd_batches=1e8, seed=1, verbose=TRUE, target.dims=2, n.cores=self$n.cores, ...) {
+    embedGraph=function(method='largeVis', name=NULL, M=1, gamma=1, alpha=0.1, perplexity=NA, sgd_batches=1e8, seed=1, verbose=TRUE, target.dims=2, n.cores=self$n.cores, ...) {
       supported.methods <- c('largeVis', 'UMAP')
       if(!method %in% supported.methods) { stop(paste0("currently, only the following embeddings are supported: ",paste(supported.methods,collapse=' '))) }
 
+      if (is.null(name)) {
+        name <- method
+        if(is.null(name)) {
+          name <- "embedding"
+        }
+      }
+
       if (method == 'largeVis') {
         wij <- as_adj(self$graph,attr='weight');
         if(!is.na(perplexity)) {
           wij <- conos:::buildWijMatrix(wij,perplexity=perplexity,threads=n.cores)
         }
         coords <- conos:::projectKNNs(wij = wij, dim=target.dims, verbose = verbose,sgd_batches = sgd_batches,gamma=gamma, M=M, seed=seed, alpha=alpha, rho=1, threads=n.cores)
         colnames(coords) <- V(self$graph)$name
-        self$embedding <- t(coords);
+        if (!is.null(self$embedding)){
+          self$embedding <- t(coords)
+        } else{
+          embedding.result <- t(coords)
+        }
       } else {
-        if (!requireNamespace("uwot", quietly=T))
+        if (!requireNamespace("uwot", quietly=TRUE))
           stop("You need to install package 'uwot' to be able to use UMAP embedding.")
 
-        self$embedding <- embedGraphUmap(self$graph, verbose=verbose, return.all=F, n.cores=n.cores, target.dims=target.dims, ...)
+        embedding.result <- embedGraphUmap(self$graph, verbose=verbose, return.all=FALSE, n.cores=n.cores, target.dims=target.dims, ...)
```

Comment:
So, now `self$embedding` is used only for largeVis, but not for UMAP :) It should be consistent.

Reply:
Conceptual mistakes on my part; let's regroup below.